### PR TITLE
Add stroking to polygons

### PR DESCRIPTION
**Pull request by manthey**
This is done by creating a sub-feature of lines for the polygons if the polygon feature has any value other than false for the stroke style (either true or a function).  Lines are created for each exterior and interior loop in the polygons, and are not set for interaction.  Lines are created with the same starting and ending point.

Strokes can have all of the properties that lines get -- strokeWidth, strokeColor, strokeOpacity, strokeStyle.  Lines are rendered with the same renderer as the polygons (and on the same level).

Polygons now have a flag to turn off filling the polygon (though the polygon is still created and still earcut, so this doesn't speed things up for polygons with fill = false).

It would be better to add a flag to the line feature to indicate a closed loop.

Changed the gl.object class to not re-subclass sceneObject, as all gl features are already subclasses of sceneObject.

This requires a vgl update to allow unbinding vertex data.  That change requires an update to the quad feature.

Fixed a typo in the lineFeature.

This resolves issue #494.

### CODE REVIEW

**Comment by manthey** [comment]
This is only a WIP until a new VGL version is available.

**Comment by jbeezley** [comment]
This looks good, I just need to fix a couple of the selenium tests before it goes in.

**Comment by jbeezley** [comment]
I found out the changes here broke styling in the `jsonReader` class because the call signature of the `stroke` and `fill` styles changed to be per polygon rather than per vertex.  The change makes sense because you can't interpolate a boolean.

Anyway, I updated the `glMultiPolygon` test to add strokes and fixed a couple of other issues that were causing spurious test failures.  With these changes, it LGTM.

**Comment by jbeezley** [comment]
One other thing I noticed that we should look into after this goes in is the quality of the stroking, which at the moment is pretty bad.  See the screenshot attached, which is at display resolution (not zoomed in).

1. Many line segments seem to be missing triangles.  This could be due to numerical precision.
2. It almost looks like the fill bleeds out beyond the stroke, but it could be some weird antialiasing effect.

screen shot 2016-07-16 at 1 24 37 pm

**Comment by manthey** [comment]
I filed issue #596 about the way lines are rendered.  They are rendered after the fill, so should never underneath it.  You'll see the exact same lines (and same errors) with a fill opacity of 0.  It appears like the vertices of some of the triangles are wrong due to an error in how mitering is calculated.

**Comment by manthey** [comment]
@jbeezley Can you check if the selenium tests need to be updated after the line fix?

**Comment by jbeezley** [comment]
Done.  LGTM.

File: src/gl/lineFeature.js [hunk 1]
```diff
@@ -96,6 +94,8 @@ var gl_lineFeature = function (arg) {
           '  else  anglePrev = atan(deltaPrev.y / aspect, deltaPrev.x);',
           '  if (deltaNext.xy == vec2(0.0, 0.0)) angleNext = anglePrev;',
           '  float angle = (anglePrev + angleNext) / 2.0;',
+          '  if (abs(anglePrev - angleNext) >= PI)',
+          '    angle += PI;',
```
Comment (aashish24) [inline src/gl/lineFeature.js]: +1 :+1:

**Pull request merged**